crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -347,11 +347,11 @@ impl AcirContext {
     }
 
     /// Adds a new variable that is constrained to be the logical NOT of `x`.
-    ///
-    /// `x` must be a 1-bit integer (i.e. a boolean)
-    pub(crate) fn not_var(&mut self, x: AcirVar) -> AcirVar {
-        // Since `x` can only be 0 or 1, we can derive NOT as 1 - x
-        self.add_data(AcirVarData::Expr(&Expression::one() - self.vars[x].to_expression().as_ref()))
+    pub(crate) fn not_var(&mut self, x: AcirVar, typ: AcirType) -> Result<AcirVar, AcirGenError> {
+        let bit_size = typ.bit_size();
```

Comment:
Does this work for signed integers?

Reply:
You would need the binary representation of 1111111111 as a signed integer which is -1 , where the top bit denotes the sign